app/code/Magento/Catalog/Model/ImageUploader.php

```diff
@@ -46,145 +47,66 @@ class ImageUploader
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager
      * @param \Psr\Log\LoggerInterface $logger
-     * @param string $baseTmpPath
-     * @param string $basePath
-     * @param string[] $allowedExtensions
-     * @param string[] $allowedMimeTypes
+     * @param Provider $provider
+     * @param Name $fileNameLookup
+     * @throws \Magento\Framework\Exception\FileSystemException
      */
     public function __construct(
         \Magento\MediaStorage\Helper\File\Storage\Database $coreFileStorageDatabase,
         \Magento\Framework\Filesystem $filesystem,
         \Magento\MediaStorage\Model\File\UploaderFactory $uploaderFactory,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
         \Psr\Log\LoggerInterface $logger,
-        $baseTmpPath,
-        $basePath,
-        $allowedExtensions,
-        $allowedMimeTypes = []
+        Provider $provider,
+        Name $fileNameLookup
     ) {
         $this->coreFileStorageDatabase = $coreFileStorageDatabase;
         $this->mediaDirectory = $filesystem->getDirectoryWrite(\Magento\Framework\App\Filesystem\DirectoryList::MEDIA);
         $this->uploaderFactory = $uploaderFactory;
         $this->storeManager = $storeManager;
         $this->logger = $logger;
-        $this->baseTmpPath = $baseTmpPath;
-        $this->basePath = $basePath;
-        $this->allowedExtensions = $allowedExtensions;
-        $this->allowedMimeTypes = $allowedMimeTypes;
-    }
-
-    /**
-     * Set base tmp path
-     *
-     * @param string $baseTmpPath
-     *
-     * @return void
-     */
-    public function setBaseTmpPath($baseTmpPath)
-    {
-        $this->baseTmpPath = $baseTmpPath;
-    }
-
-    /**
-     * Set base path
-     *
-     * @param string $basePath
-     *
-     * @return void
-     */
-    public function setBasePath($basePath)
-    {
-        $this->basePath = $basePath;
+        $this->provider = $provider;
+        $this->fileNameLookup = $fileNameLookup;
     }
 
     /**
-     * Set allowed extensions
-     *
-     * @param string[] $allowedExtensions
+     * Retrieve path
      *
-     * @return void
-     */
-    public function setAllowedExtensions($allowedExtensions)
-    {
-        $this->allowedExtensions = $allowedExtensions;
-    }
-
-    /**
-     * Retrieve base tmp path
+     * @param string $path
+     * @param string $imageName
      *
      * @return string
      */
-    public function getBaseTmpPath()
+    public function getFilePath($path, $imageName)
     {
-        return $this->baseTmpPath;
+        return rtrim($path, '/') . '/' . ltrim($imageName, '/');
     }
 
     /**
-     * Retrieve base path
+     * Get the base path from the config provider
      *
      * @return string
      */
     public function getBasePath()
     {
-        return $this->basePath;
-    }
-
-    /**
-     * Retrieve allowed extensions
-     *
-     * @return string[]
-     */
-    public function getAllowedExtensions()
```

Comment:
Is 2.4 a breaking change? Am I allowed to remove these? I could not see any usaged in place for them apart from in the class.